.semaphore/semaphore.yml

```diff
@@ -1,68 +1,82 @@
-            - flutter pub get flutter-packages
+            - flutter pub get
             - 'cache store flutter-packages-$SEMAPHORE_GIT_BRANCH-$(checksum pubspec.yaml),flutter-packages-$(checksum pubspec.yaml),flutter-packages /root/.pub-cache'
-  - name: Linting
+      prologue:
+        commands:
+          - checkout
+          - 'cache restore flutter-packages-$SEMAPHORE_GIT_BRANCH-$(checksum pubspec.yaml),flutter-packages-$(checksum pubspec.yaml),flutter-packages'
+          - flutter pub get
+  - name: Lint
     task:
+      prologue:
+        commands:
+          - checkout
+          - 'cache restore flutter-packages-$SEMAPHORE_GIT_BRANCH-$(checksum pubspec.yaml),flutter-packages-$(checksum pubspec.yaml),flutter-packages'
+          - flutter pub get
       jobs:
-        - name: Format code
+        - name: Format
           commands:
             - flutter format --set-exit-if-changed .
-        - name: Run code lint
+        - name: Analyze
           commands:
             - flutter analyze .
+  - name: Unit and widget tests
+    task:
       prologue:
         commands:
           - checkout
           - 'cache restore flutter-packages-$SEMAPHORE_GIT_BRANCH-$(checksum pubspec.yaml),flutter-packages-$(checksum pubspec.yaml),flutter-packages'
-  - name: Tests
-    task:
+          - flutter pub get
       jobs:
-        - name: Run unit and widget tests
+        - name: Test
           commands:
-            - 'flutter test test '
+            - flutter test test
+  - name: Run UI tests
+    task:
       prologue:
         commands:
           - checkout
           - 'cache restore flutter-packages-$SEMAPHORE_GIT_BRANCH-$(checksum pubspec.yaml),flutter-packages-$(checksum pubspec.yaml),flutter-packages'
-  - name: Integration Tests
-    task:
+          - flutter pub get
+          - device_uuid=$(xcrun simctl create ios-simulator com.apple.CoreSimulator.SimDeviceType.iPhone-11-Pro com.apple.CoreSimulator.SimRuntime.iOS-14-5)
+          - xcrun simctl boot $device_uuid
       jobs:
-        - name: Add New Item Test
+        - name: Add new item
           commands:
             - flutter test integration_test/add_new_todo_item_test.dart
-        - name: Edit Existing Item Test
+        - name: Edit exist item
           commands:
             - flutter test integration_test/edit_existing_item_test.dart
-        - name: Delete Existing Item Test
+        - name: Delete item
           commands:
             - flutter test integration_test/delete_existing_item_test.dart
+    run:
+      when: 'change_in(''/lib'', {default_branch: ''main'', pipeline_file: ''ignore''}))'
+  - name: Build artifact
+    task:
       prologue:
         commands:
           - checkout
           - 'cache restore flutter-packages-$SEMAPHORE_GIT_BRANCH-$(checksum pubspec.yaml),flutter-packages-$(checksum pubspec.yaml),flutter-packages'
-          - sdkmanager "platform-tools" "platforms;android-29" "build-tools;30.0.0" "emulator"
```

Comment:
Is this passing? The following open an emulator to run the integration tests.

Reply:
this line isn't necessary as this semaphore.yml file is for an iOS build